clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java

```diff
@@ -296,36 +314,6 @@ public AuthenticationException authenticationException(String id) {
         return state != null ? state.authenticationException : null;
     }
 
-    /**
-     * Resets the failure count for a node and sets the reconnect backoff to the base
-     * value configured via reconnect.backoff.ms
-     *
-     * @param nodeState The node state object to update
-     */
-    private void resetReconnectBackoff(NodeConnectionState nodeState) {
```

Comment:
Can't we keep this method to perform reset (perhaps rename the method) and include all types of reset?

Reply:
Including all types of reset together is probably not a good choice because the reset of failed attempts and the reset of the connection timeout may happen in different places. 

However, I agree we should have some abstraction on the update and reset logic. I'll put the logic in new class methods.